extractor/src/main/kotlin/de/snowii/extractor/extractors/Tests.kt

```diff
@@ -0,0 +1,146 @@
+import net.minecraft.world.gen.noise.NoiseConfig
+
+import java.lang.reflect.Method
+import java.util.Arrays
+import kotlin.reflect.full.createType
+import kotlin.reflect.full.declaredFunctions
+import kotlin.reflect.jvm.javaMethod
+import kotlin.reflect.KFunction
+
+class Tests : Extractor.Extractor {
+    override fun fileName(): String = "chunk.json"
+
+	private fun createFluidLevelSampler(settings: ChunkGeneratorSettings): AquiferSampler.FluidLevelSampler {
+		val fluidLevel = AquiferSampler.FluidLevel(-54, Blocks.LAVA.getDefaultState());
+		val i = settings.seaLevel();
+		val fluidLevel2 = AquiferSampler.FluidLevel(i, settings.defaultFluid());
+		return AquiferSampler.FluidLevelSampler {_, y, _ -> if (y < Math.min(-54, i)) fluidLevel else fluidLevel2};
+	}
+
+	private fun get_index(config: GenerationShapeConfig, x: Int, y: Int, z: Int): Int {
+		if (x < 0 || y < 0 || z < 0) {
+			System.err.println("Bad local pos");
+			System.exit(1);
+		}
+		return config.height() * 16 * x + 16 * y + z
+	}
+
+	// This is basically just what NoiseChunkGenerator is doing
+	private fun populate_noise(start_x: Int, start_z: Int, sampler: ChunkNoiseSampler, config: GenerationShapeConfig, settings: ChunkGeneratorSettings): IntArray? {
+		val result = IntArray(16 * 16 * config.height())
+
+		for (method: KFunction<*> in sampler::class.declaredFunctions) {
+			if (method.name.equals("sampleBlockState")) {
+				sampler.sampleStartDensity()
+				val k = config.horizontalCellBlockCount()
+				val l = config.verticalCellBlockCount()
+
+				val m = 16 / k
+				val n = 16 / k
+
+				val cellHeight = config.height() / l
+				val minimumCellY = config.minimumY() / l
+
+				for (o in 0..<m) {
+					sampler.sampleEndDensity(o)
+					for (p in 0..<n) {
+						for (r in (0..<cellHeight).reversed()) {
+							sampler.onSampledCellCorners(r, p)
+							for (s in (0..<l).reversed()) {
+								val t = (minimumCellY + r) * l + s
+								val d = s.toDouble() / l.toDouble()
+								sampler.interpolateY(t, d)
+								for (w in 0..<k) {
+									val x = start_x + o * k + w
+									val y = x and 15
+									val e = w.toDouble() / k.toDouble()
+									sampler.interpolateX(x, e)
+									for (z in 0..<k) {
+										val aa = start_z + p * k + z
+										val ab = aa and 15
+										val f = z.toDouble() / k.toDouble()
+										sampler.interpolateZ(aa, f)
+										var blockstate = method.call(sampler) as BlockState?
+										if (blockstate == null) {
+											blockstate = settings.defaultBlock()
+										}
+										val index = this.get_index(config, y, t - config.minimumY(), ab)
+										result[index] = Block.getRawIdFromState(blockstate)
+									}
+								}
+							}
+						}
+					}
+					sampler.swapBuffers()
+				}
+				sampler.stopInterpolation()
+				return result
+			}
+		}
+		System.err.println("No valid method found for block state sampler!");
+		return null;
+	}
+
+	override fun gson_builder(): Gson {
+		return GsonBuilder().create()
+	}
```

Comment:
Is the only reason your using this is that you don't want pretty printing ?. Feel free to remove pretty printing from everything

Reply:
@Snowiiii removed